### ISSUE TITLE
Incorrect error message for use-before-declaration on const enum in `preserveConstEnums`

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#38722]
```ts
// @preserveConstEnums: true
E.A;

const enum E { A }
```

Expected:

```
Enum 'E' used before its declaration
```

Actual:

```
Class 'E' used before its declaration
```

**Comment by DanielRosenwasser** [comment]
No new tests necessary, should already be covered by `tests/cases/compiler/blockScopedEnumVariablesUseBeforeDef_preserve.ts`.

**Comment by DanielRosenwasser** [comment]
Thanks @a-tarasyuk!
